flight/PiOS/Common/pios_dsm.c

```diff
@@ -316,9 +316,37 @@ int32_t PIOS_DSM_Init(uintptr_t *dsm_id,
 	/* Bind the configuration to the device instance */
 	dsm_dev->cfg = cfg;
 
+	uint8_t num_pulses = 0;
+
+	// check user settings to determine which resolution mode to use
+	switch (mode)
+	{
+	case HWSHARED_DSMXMODE_AUTODETECT:
+		dsm_dev->resolution = DSM_UNKNOWN;
+		break;
+	case HWSHARED_DSMXMODE_FORCE10BIT:
+		dsm_dev->resolution = DSM_10BIT;
+		break;
+	case HWSHARED_DSMXMODE_FORCE11BIT:
+		dsm_dev->resolution = DSM_11BIT;
+		break;
+	case HWSHARED_DSMXMODE_BIND3PULSES:
+	case HWSHARED_DSMXMODE_BIND4PULSES:
+	case HWSHARED_DSMXMODE_BIND5PULSES:
+	case HWSHARED_DSMXMODE_BIND6PULSES:
+	case HWSHARED_DSMXMODE_BIND7PULSES:
+	case HWSHARED_DSMXMODE_BIND8PULSES:
+	case HWSHARED_DSMXMODE_BIND9PULSES:
+	case HWSHARED_DSMXMODE_BIND10PULSES:
+		num_pulses = 3 + mode - HWSHARED_DSMXMODE_BIND3PULSES;
+		break;
+	default:
```

Comment:
Just an FYI:  I'm ambivalent about default: clauses in switch statements.  They may catch (are not required to catch per C99 standard) when invalid values are passed in.  But they also squelch a compiler warning about not naming all enum values.